File: src/main/java/net/dashflight/data/redis/DashflightRedisConnectionOptionProvider.java
package net.dashflight.data.redis;

import net.dashflight.data.config.ConfigValue;
import net.dashflight.data.config.Configurable;

/**
 * Pulls the required redis connection options from s3
 */
public class DashflightRedisConnectionOptionProvider implements RedisConnectionOptionProvider, Configurable {

    private static final String APP_NAME = "redis";


    @ConfigValue("redis_host")
    private String host;

    @ConfigValue("redis_port")
    private int port;

    @ConfigValue("max_pool_size")
    private int maxPoolSize = 4;


    public DashflightRedisConnectionOptionProvider() {
        registerWith(APP_NAME);
    }


    @Override
    public RedisConnectionOptions get() {
        return RedisConnectionOptions.builder()
                .host(host)
                .port(port)
                .maxPoolSize(maxPoolSize)
                .build();
    }
}

File: src/main/java/net/dashflight/data/redis/DashflightRedisConnectionOptionProvider.kt
package net.dashflight.data.redis

import hydro.engine.Hydro.hydrate

/**
 * Pulls the required redis connection options from s3
 */
class DashflightRedisConnectionOptionProvider : RedisConnectionOptionProvider {

    private val host: String by hydrate("redis_host")

    private val port: Int by hydrate("redis_port")

    private val maxPoolSize: Int by hydrate("max_pool_size", 4)

    override fun get(): RedisConnectionOptions {
        return RedisConnectionOptions(host, port, maxPoolSize)
    }
}